src/Misc/layoutroot/safe_sleep.sh

```diff
@@ -1,6 +1,15 @@
 #!/bin/bash
 
 SECONDS=0
-while [[ $SECONDS != $1 ]]; do
+
+# Bash 4.0 and above supports read.
+if [[ -n "$BASH_VERSINFO" && "${BASH_VERSINFO[0]}" -ge 4 ]]; then
+    echo foo
+    # Bash has no sleep builtin, but read with a timeout can behave in same way.
+    read -rt "$1" <> <(:) || :
+fi
+# Fallback to busy wait.
+while [[ $SECONDS -lt $1 ]]; do
```

Comment:
[`-lt` is vulnerable to code-injection](https://mywiki.wooledge.org/BashProgramming/05#Arithmetic_Expansion) ([SC](https://github.com/koalaman/shellcheck/issues/3067))

Previously [mentioned here](https://github.com/actions/runner/pull/3157#issuecomment-3638892152)